components/authz-service/server/v2/rules_property_test.go

```diff
@@ -0,0 +1,124 @@
+package v2_test
+
+import (
+	"context"
+	"reflect"
+	"testing"
+
+	"github.com/leanovate/gopter"
```

Comment:
We have used one data generator in our code base already, https://github.com/jaswdr/faker. Is gopter more useful?

Reply:
> Is gopter more useful?

100 times _yes_ -- since it's not _just_ a data generator. Back when we've introduced faker, you might remember our conversations (also with @phiggins, https://github.com/chef/a2/pull/4294 and previous gopter usage introduced earlier: https://github.com/chef/a2/pull/4369): fake data is nice for tests, but the icing on the cake is _property based tests_. This is what gopter gives us: a way to state **properties**, and assert them using a large battery of random inputs.

I.e. instead of
> generate one random policy, see that it can be created

we formulate a property
> when CreateRule is called with some input, it will lead to a rule in storage

and have the framework try that a hundred times with different inputs.

**Also**, any decent PBT framework gives you _shrinking_: when it finds an input such that the property _not holds_, it attempts to find the smallest input that does the same thing. It's vital for dealing with random inputs. 

Now, this doesn't mean what we don't need unit tests. And it also doesn't mean that the way we're using it here is the one and only correct way.